File: Python/DecryptSubstitution_German.py
def DecryptSubstitution_German(ciphertext_ascii_30, key_list):
    lookup_key = [((key - 97), (i + 97)) for i, key in enumerate(key_list)]
    return [
        j[:][1] for i in ciphertext_ascii_30 for j in lookup_key
        if (i - 65) == j[:][0]
    ]

File: Julia/DecryptSubstitution_German.jl
function DecryptSubstitution_German(ciphertext_ascii_30, key_list)
    dectext_ascii_30 = ciphertext_ascii_30 - 64
    for (i, key) in enumerate(key_list)
        dectext_ascii_30[find(x -> x == key - 96, dectext_ascii_30)] = i + 96
    end
    return dectext_ascii_30
end